### PR TITLE
Functional port expose. Not a full port mapping implementation.

### PR DESCRIPTION
**Pull request by erikh**
Docker-DCO-1.1-Signed-off-by: Erik Hollensbe github@hollensbe.org (github: erikh)

### CODE REVIEW

**Comment by icecrime** [comment]
I'm waiting for your "go" to start reviewing as I heard you say there might be an issue with the patch.

**Comment by erikh** [comment]
Yeah, hang on a bit.

**Comment by tiborvass** [comment]
Why don't we close the PR if it's not ready? ;)

**Comment by erikh** [comment]
It's ready.

File: daemon/create.go [hunk 1]
```diff
@@ -174,7 +174,8 @@ func (daemon *Daemon) Create(config *runconfig.Config, hostConfig *runconfig.Hos
 	// Expose ports on the new endpoint
 	if c.Config.ExposedPorts != nil {
 		for port, _ := range config.ExposedPorts {
-			if err := ep.Expose(port.String(), false); err != nil {
+			// DEMO FIXME make publish selectable
+			if err := ep.Expose(port.String(), true); err != nil {
```
Comment (icecrime) [inline daemon/create.go]: I'm surprised this is required:
- There's this first loop on **exposed** ports which does `e.Expose(..., false)`
- There's a second loop right below on **published** ports which does `e.Expose(..., true)`

As long as the endpoint handles being called twice on a same port specification sequentially with `publish=false` then `publish=true`, this change should not be necessary.

Comment (erikh) [inline daemon/create.go]: well I just ignore the argument now, so what I'll do is change this back, then just bail immediately on non-publish ports.

**Comment by erikh** [comment]
fixed

**Pull request merged**